bpf/process/types/basic.h

```diff
@@ -1926,9 +1930,20 @@ rate_limit(__u64 ratelimit_interval, struct msg_generic_kprobe *e)
 	ro_heap = map_lookup_elem(&ratelimit_ro_heap, &zero);
 
```

Comment:
would it make sense to memset zero those returned heap entries? (didn't check all the logic)

Reply:
I overwrite the heap with a read only heap that is all zero, using a probe_read. It should cost about the same as a memset but doesn't cost me the instruction / complexity count.